File: 01x10/c-004/main.php
<?php
function isPalindrome($num) {
    return $num == strrev($num);
}

function largestPalindrome($num) {
    $result = 0;
    for ($i = $num; $i > 0; $i--) {
        if (isPalindrome($num * $i)) {
            $result = $num * $i;
            break;
        }
    }
    return $num > 0 ? max($result, largestPalindrome($num - 1)) : 0;
}

print largestPalindrome(999);

File: 01x10/c-004/main.rs
use std::cmp;

fn is_palindrome(num: i32) -> bool {
    let strnum = num.to_string();
    return strnum == strnum
        .chars()
        .rev()
        .collect::<String>();
}

fn largest_palindrome(num: i32) -> i32 {
    let mut result = 0;
    for i in (0..num).rev() {
        if is_palindrome(num * i) {
            result = num * i;
            break;
        }
    }
    return if num > 0 {
        cmp::max(result, largest_palindrome(num-1))
    } else {
        0
    }
}

fn main() {
    println!("{}", largest_palindrome(999));
}
